### PR TITLE
Bug 1155104 - Show a sensible error message when trying to access speed dial numbers without a SIM card r=drs

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=76982295bedb8dbcc9181ceb02c8b725a70eac04

File: apps/communications/dialer/test/unit/keypad_test.js [hunk 1]
```diff
@@ -846,6 +846,17 @@ suite('dialer/keypad', function() {
             assert.equal(number, numbers[0]);
           }).then(done, done);
         });
+
+        test('An error message is shown when no SIM card is present',
+        function(done) {
+          this.sinon.stub(MockIccManager.prototype, 'getIccById').returns(null);
+
+          subject._getSpeedDialNumber(speedDialIndex).then(function() {
```
Comment (DouglasSherk) [inline apps/communications/dialer/test/unit/keypad_test.js]: Really liking this new style of unit tests, which are actually unit tests.

Comment (gabrielesvelto) [inline apps/communications/dialer/test/unit/keypad_test.js]: Yeah, it's about time we have real unit-tests :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3d00b1445a1de5910e701b4f3f48a66250ab30b1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9574e572807c05cae472c1eff4f8f47ac39caad6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=611902a1eaecb63d51725c3cba499d84f0ffcbdd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d67971f0c3decacb3cf1fb4b7913c01dece27974

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b4237b0864412bfaeebd6eac476d9b86c49023f3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8d0855eb5238800cf33b5e3a5648ea8479ac62ff

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4ab9c25309c05c57bf8f20858cbdc4f03e6ec759

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ab797dcc955dcced80e427fe1dcfe69ee93fdc01

**Pull request merged**